org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/JavaModelEventProviderTest.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.jdt.core.ElementChangedEvent;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IElementChangedListener;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler;
+import org.eclipse.jdt.ls.core.internal.managers.AbstractProjectsManagerBasedTest;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences.Severity;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JavaModelEventProviderTest extends AbstractProjectsManagerBasedTest {
+	private DocumentLifeCycleHandler lifeCycleHandler;
+	private JavaClientConnection javaClient;
+
+	@Before
+	public void setup() throws Exception {
+		mockPreferences();
+
+		javaClient = new JavaClientConnection(client);
+		lifeCycleHandler = new DocumentLifeCycleHandler(javaClient, preferenceManager, projectsManager, false);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		javaClient.disconnect();
+		for (ICompilationUnit cu : JavaCore.getWorkingCopies(null)) {
+			cu.discardWorkingCopy();
+		}
+	}
+
+	private Preferences mockPreferences() {
+		Preferences mockPreferences = Mockito.mock(Preferences.class);
+		Mockito.when(preferenceManager.getPreferences()).thenReturn(mockPreferences);
+		Mockito.when(preferenceManager.getPreferences(Mockito.any())).thenReturn(mockPreferences);
+		Mockito.when(mockPreferences.getIncompleteClasspathSeverity()).thenReturn(Severity.ignore);
+		return mockPreferences;
+	}
+
+	@Test
+	public void testSendNotification() throws Exception {
+		JavaCore.addElementChangedListener(new IElementChangedListener() {
+
+			@Override
+			public void elementChanged(ElementChangedEvent event) {
+				client.notify("notify", event);
```

Comment:
Use a different name for the notification message here. For example: "elementChanged"